legate/util/args.py

```diff
@@ -153,25 +159,47 @@ def parse_library_command_args(
         prog=f"<{libname} program>", add_help=False, allow_abbrev=False
     )
 
-    lib_prefix = f"-{libname}:"
+    # Some explanation is in order. Argparse treats arguments with a single
+    # dash differently, e.g. "-xyz" is interpreted as "-x -y -z". This can
+    # cause confusion and clashes when there are multiple single-dash args
+    # with identical prefixes. TLDR; we want "-legate:foo" to behave just
+    # as if it was "--legate:foo". In order to do this, we configure a parser
+    # for "long argumens" and then munge the values in sys.argv to update
+    # any "short prefix" arguments to be "long prefix" arguments first, before
+    # parsing. We also take care to update any output. The alternative here
+    # would be to abandon argparse entirely, and parse sys.argv manually.
+    #
+    # ref: https://github.com/nv-legate/legate.core/issues/415
+
+    short_prefix = f"-{libname}:"
+    long_prefix = f"-{short_prefix}:"
```

Comment:
Doesn't strictly matter, but note that there will be an extra `:` in the "long" format arguments, i.e. `-legate:foo` will be converted to `--legate::foo`.

Reply:
yes, but let's not confuse future readers https://github.com/nv-legate/legate.core/pull/416/commits/451cd25472d4158487659591a49d235233191c78